### PR TITLE
feat(helm-chart): Add option to override resource requests and limits…

### LINKED ISSUE #1083: Resource requests and limits for helm CRD hooks
**Specify resource requests and limits for crd hooks in helm chart**

I would like to be able to specify resource limits and requests for the jobs that are created in the helm hooks for upgrade and keep crd jobs.

When a person uses the helm chart to install the secrets-store-csi-driver, the jobs (keep-crd and upgrade-crd) that are created does not specify any requests for the pods.

In a namespace with LimitRange defined (https://kubernetes.io/docs/concepts/policy/limit-range/) the pods will easily get oomkilled.

### PR DESCRIPTION
**Pull request by haugom**
… for helm CRD hooks

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
This PR allows users to optionally set resource request/limits on the jobs that are created from the helm chart in this repository.

If the kubernetes cluster has LimitRanges (https://kubernetes.io/docs/concepts/policy/limit-range/) enforcing default values on pods, the CRD hooks could easily run out of memory.

This PR is to allow users of the chart to be able to set resource limits on the jobs that are part of the helm hooks.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #1083

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [x] includes documentation
- [ ] adds unit tests

### CODE REVIEW

**Review by aramase — COMMENTED** [review]
/ok-to-test

File: manifest_staging/charts/secrets-store-csi-driver/README.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ The following table lists the configurable parameters of the csi-secrets-store-p
 | `linux.crds.image.repository`           | Linux crds image repository                                                                                                                                                    | `k8s.gcr.io/csi-secrets-store/driver-crds`              |
 | `linux.crds.image.pullPolicy`           | Linux crds image pull policy                                                                                                                                                   | `IfNotPresent`                                          |
 | `linux.crds.image.tag`                  | Linux crds image tag                                                                                                                                                           | `v1.2.4`                                                |
+| `linux.crds.resources`                  | The resource request/limits for the linux crds container images                                                                                                                | `{}`                                                    |
```
Comment (aramase) [inline manifest_staging/charts/secrets-store-csi-driver/README.md:40]:
```suggestion
| `linux.crds.resources`                  | The resource request/limits for the linux crds container image                                                                                                                | `{}`                                                    |
```

Comment (haugom) [reply manifest_staging/charts/secrets-store-csi-driver/README.md:40]: I agree. I will change it.

Should I also change the chart version or will your team do that when you make a new release and move it to the other folder?

Comment (aramase) [reply manifest_staging/charts/secrets-store-csi-driver/README.md:40]: The chart versions are upgraded during release.

**Review by aramase — APPROVED** [review]
/approve

i'll let @nilekhc or @tam7t `/lgtm` this after review

**Review by aramase — COMMENTED** [review]
@haugom could you rebase the PR against main to pick up changes merged to `main` branch that'll fix the CI failures?

**Comment by haugom** [comment]
Hi

I tried to rebase but it became a merge. Is this OK or does it need to be a rebase?

**Review by aramase — COMMENTED** [review]
/lgtm

/label tide/merge-method-squash

**Pull request merged**